astropy/table/table.py

```diff
@@ -870,9 +870,8 @@ def _init_from_list_of_dicts(self, data, names, dtype, n_cols, copy):
 
         # Fill the missing entries with first values
         if missing_indexes:
-            n_cols = len(names_from_data)
```

Comment:
This line was a leftover from the original commit and is not appropriate since `n_cols` is set correctly on input.

Reply:
Oops, sorry for the sloppy review (twice)